searx/engines/bing.py

```diff
@@ -93,16 +94,34 @@ def response(resp):
         results.append({'url': url, 'title': title, 'content': content})
 
     # parse results again if nothing is found yet
-    for result in eval_xpath(dom, '//li[@class="b_algo"]'):
+
+    url_to_resolve = []
+    url_to_resolve_index = []
+    for i, result in enumerate(eval_xpath(dom, '//li[@class="b_algo"]')):
 
         link = eval_xpath(result, './/h2/a')[0]
         url = link.attrib.get('href')
         title = extract_text(link)
         content = extract_text(eval_xpath(result, './/p'))
 
+        # get the real URL either using the URL shown to user or following the Bing URL
+        if url.startswith('https://www.bing.com/ck/a?'):
+            url_cite = extract_text(eval_xpath(result, './/div[@class="b_attribution"]/cite'))
+            if not url_cite.startswith('https://') or url_cite.endswith('...'):
+                url_to_resolve.append(url.replace('&ntb=1', '&ntb=F'))
+                url_to_resolve_index.append(i)
+                url = None  # remove the result if the HTTP Bing redirect raise an exception
+            else:
+                url = url_cite
+
         # append result
         results.append({'url': url, 'title': title, 'content': content})
 
+    # follow HTTP Bing URL and update the result URLs
+    for i, redirect_response in enumerate(http_get(url_to_resolve, allow_redirects=False)):
+        if not isinstance(redirect_response, Exception):
+            results[url_to_resolve_index[i]]['url'] = redirect_response.headers['location']
```

Comment:
> can @return42 @mrpaulblack @tiekoetter @unixfox or anyone try this PR ?

The patch itself works but I expect a lot of IO by this loop: when I query about a resource with long URLs e.g. `!bing heise.de bug :de` I see a lot of HTTP requests fired to bing.com .. IDK an other solution but I think fire one HTTP request for one link is to much .. IDK how long it takes bing blocks such requests and bans the server IP (CAPTCHA).

Since I see also other issues [1], personally I tend to drop the bing-web engine.

[1] On my server I need to repeat each query twice to get results, otherwise I get bad results similar to https://github.com/searxng/searxng/issues/941

Reply:
> Since I see also other issues [1], personally I tend to drop the bing-web engine.

May be the best solution would be disable bing by default and enable duckduckgo back, because, AFAIK, DDG uses bing?